gprofiler/metadata/system_metadata.py

```diff
@@ -132,7 +134,37 @@ def get_mac_address() -> str:
         address = ":".join(["%02X" % i for i in mac])
         return address
 
-    return "unknown"
+    return UNKNOWN_VALUE
+
+
+def get_cpu_info() -> Tuple[str, str]:
+    """
+    Parse /proc/cpuinfo to get model name & flags.
+    """
+    try:
+        with open("/proc/cpuinfo") as f:
+            model_names = []
+            flags = []
+            for line in f:
+                m = re.match(r"^((?:model name)|(?:flags))[ \t]*: (.*)$", line)
+                if m is not None:
+                    field, value = m.groups()
+                    if field == "model name":
+                        model_names.append(value)
+                    else:
+                        assert field == "flags", f"unexpected field: {field!r}"
+                        flags.append(value)
+
+        if len(set(model_names)) != 1:
+            logger.warning(f"CPU model names differ between cores, reporting only the first: {model_names}")
```

Comment:
Where do we ever see this?

Reply:
IDK, where have you ever not seen it? :)

Expect the unexpected